presto-main/src/main/java/com/facebook/presto/sql/Optimizer.java

```diff
@@ -105,12 +108,17 @@ public Plan validateAndOptimizePlan(PlanNode root, PlanStage stage)
                     throw new PrestoException(QUERY_PLANNING_TIMEOUT, String.format("The query optimizer exceeded the timeout of %s.", getQueryAnalyzerTimeout(session).toString()));
                 }
                 long start = System.nanoTime();
-                root = optimizer.optimize(root, session, TypeProvider.viewOf(variableAllocator.getVariables()), variableAllocator, idAllocator, warningCollector);
-                requireNonNull(root, format("%s returned a null plan", optimizer.getClass().getName()));
+                PlanNode newRoot = optimizer.optimize(root, session, TypeProvider.viewOf(variableAllocator.getVariables()), variableAllocator, idAllocator, warningCollector);
+                requireNonNull(newRoot, format("%s returned a null plan", optimizer.getClass().getName()));
                 if (enableVerboseRuntimeStats) {
                     session.getRuntimeStats().addMetricValue(String.format("optimizer%sTimeNanos", optimizer.getClass().getSimpleName()), NANO, System.nanoTime() - start);
                 }
+                collectOptimizerInformation(optimizer, root, newRoot);
+                root = newRoot;
             }
+
+            List<PlanOptimizerInformation> planOptimizerInformationList = session.getOptimizerInformationCollector().getOptimizationInfo();
+            System.out.println("Done optimizing, invoked: " + planOptimizerInformationList.size() + " optimizations");
```

Comment:
Is this a debug log which you forgot to remove? If not, maybe consider adding a logger instead of doing System.out.println?

Reply:
will remove